hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ResourceAttributeEnricher.java

```diff
@@ -0,0 +1,74 @@
+package org.hypertrace.traceenricher.enrichment.enrichers;
+
+import static org.hypertrace.traceenricher.util.EnricherUtil.getResourceAttribute;
+
+import com.typesafe.config.Config;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.hypertrace.core.datamodel.AttributeValue;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.traceenricher.enrichment.AbstractTraceEnricher;
+import org.hypertrace.traceenricher.enrichment.clients.ClientRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ResourceAttributeEnricher extends AbstractTraceEnricher {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ResourceAttributeEnricher.class);
+  private static final String RESOURCE_ATTRIBUTES_CONFIG_KEY = "attributes";
+  private static final String NODE_SELECTOR_KEY = "node.selector";
+  private List<String> resourceAttributesToAdd = new ArrayList<>();
+
+  public List<String> getResourceAttributesToAdd() {
+    return resourceAttributesToAdd;
+  }
+
+  @Override
+  public void init(Config enricherConfig, ClientRegistry clientRegistry) {
+    if (enricherConfig.hasPath(RESOURCE_ATTRIBUTES_CONFIG_KEY)) {
+      resourceAttributesToAdd = enricherConfig.getStringList(RESOURCE_ATTRIBUTES_CONFIG_KEY);
+    }
+  }
+
+  private boolean isValidEvent(Event event) {
+    if (resourceAttributesToAdd.isEmpty()) {
```

Comment:
If there are no resource attributes to add, then remove this enricher from the Enricher DAG? This logic should be moved to when Trace Enricher is bootstrapping.